text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1601 @@
+
+## Data Model
+
+This section describes various protobuf messages that are used to represent profiles data.
+
+### Relationships Diagram
+
+The following diagram shows the relationships between the messages. Relationships between messages are represented by either embedding one message in another (red arrows), or by referencing a message by index in a lookup table (blue arrows). More on that in [Relationships Between Messages](#relationships-between-messages) section below.
+
+In addition to that, relationship between `samples` and `locations` is further optimized for better performance. More on that in [Relationship Between Samples and Locations](#relationship-between-samples-and-locations) section below.
+
+![diagram of data relationships](./images/otep0239/profiles-data-model.png)
+
+### Relationships With Other Signals
+
+There are two types of relationships between profiles and other signals:
+
+* from other signals to profiles (e.g from log records, exemplars or trace spans)
+* from profiles to other signals
+
+#### From Profiles to Other Signals
+
+[Link](#message-link) is a message that is used to represent connections between profile [Samples](#message-sample) and trace spans. It uses `trace_id` and `span_id` as identifiers.
+
+For other signals, such as logs or metrics, because other signals use the same way of linking between such signals and traces (`trace_id` and `span_id`), it is possible to correlate profiles with other signals using this same information.
+
+#### From Other Signals to Profiles
+
+Other signals can use `profile_id` to reference a profile. For example, a log record can reference a profile that was collected at the time when the log record was generated by using `profile_id` as one of the attributes. This allows to correlate logs with profiles.
+
+Additionally, `trace_id`, `span_id` can be used to reference specific [Samples](#message-sample) in a Profile, since [Samples](#message-sample) are linked to traces with these same identifiers using [Links](#message-link).
+
+The exact details of such linking are out of scope for this OTEP. It is expected that the exact details will be defined in Profiles part of [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification).
+
+### Compatibility With Original pprof
+
+The proposed data model is backwards compatible with original pprof in a sense that a pprof file generated by existing software can be parsed using the new proto.
```

Comment:
Can we further clarify this? Does parsing existing pprof file result in all of its data fully present in the fields of the new proto? The reason I am asking is that it in Protobufs it is valid to parse data that contains unknown fields and it is not an error, however essentially doing so loses data.

Does new proto define all fields of original pprof such that parsing it is a looseless operation?

Reply:
Yes, we preserved all fields so that old pprof files can still be parsed using the new proto, and no data is lost. I'll clarify this in the OTEP too.